src/lib/gallery.ts

```diff
@@ -0,0 +1,192 @@
+import { get as getStore } from 'svelte/store'
+import * as wn from 'webnative'
+import { filesystemStore, galleryStore } from '../stores'
+import { convertUint8ToString } from '$lib/common/utils'
+import { addNotification } from '$lib/notifications'
+
+export enum AREAS {
+  PUBLIC = 'Public',
+  PRIVATE = 'Private',
+}
+
+export type Image = {
+  cid: string
+  ctime: number
+  name: string
+  private: boolean
+  size: number
+  src: string
+}
+
+export type Gallery = {
+  publicImages: Image[] | null
+  privateImages: Image[] | null
+  selectedArea: AREAS
+  loading: boolean
+}
+
+const GALLERY_DIRS = {
```

Comment:
Where are the gallery directories created? If we are missing this, we might want to create them right after bootstrapping the filesystem and setting the filesystem store:

https://github.com/fission-codes/webnative-app-template/blob/eec81aa1bf1ec03a4ac5ab8e433a8107e86956aa/src/lib/common/webnative.ts#L91-L92

Reply:
> Where are the gallery directories created? If we are missing this, we might want to create them right after bootstrapping the filesystem and setting the filesystem store:
> 
> https://github.com/fission-codes/webnative-app-template/blob/eec81aa1bf1ec03a4ac5ab8e433a8107e86956aa/src/lib/common/webnative.ts#L91-L92

Good question! I forgot to ask you about that. I wasn't sure if it was meant to work differently in AOL apps. When I push an image to either file system for the first time, the directories are automatically created, so I wasn't sure if i was supposed to create them beforehand 😅 I'll add it in here though, thanks!

Reply:
Yeah, that's a bit different in AOL apps. In existing webnative apps, the auth lobby creates directories for you based on the permissions used at initialization. In AOL apps, these need to be created manually.

Reply:
> Yeah, that's a bit different in AOL apps. In existing webnative apps, the auth lobby creates directories for you based on the permissions used at initialization. In AOL apps, these need to be created manually.

@bgins follow-up question: what's the best way to do this in an AOL app? Should i still do it inside a [wn.initialize](https://guide.fission.codes/developers/webnative/auth) call? Or possibly another `wn` method? 

<img width="450" alt="image" src="https://user-images.githubusercontent.com/1179291/186536268-bc906d78-5543-439d-85a6-379d54fceb93.png">

Reply:
The way to do this is with the filesystem with `fs.mkdir`. I think it would be something like this:

```typescript
await fs.mkdir(webnative.path.directory('public', 'gallery'))
await fs.mkdir(webnative.path.directory('private', 'gallery'))
```

Reply:
> The way to do this is with the filesystem with `fs.mkdir`. I think it would be something like this:
> 
> ```ts
> await fs.mkdir(webnative.path.directory('public', 'gallery'))
> await fs.mkdir(webnative.path.directory('private', 'gallery'))
> ```

ohh gotcha, kk thanks!

Reply:
@bgins updated 👍🏼